crates/runix/src/command_line/mod.rs

```diff
@@ -231,6 +231,14 @@ impl NixCommandLine {
 
         M::run(&mut command).await
     }
+
+    // Set the global Nix config via the environment variables in flox.default_args so that
+    // subprocesses called by `flox` (e.g. `parser-util`) can inherit them.
+    pub fn export_config_env_vars(&self) {
+        for (name, value) in self.defaults.environment.iter() {
+            std::env::set_var(name, value);
+        }
+    }
```

Comment:
I'd rather keep this to flox/the cosumer of the library to set up

Reply:
I don't really agree. The actual data that you need for this operation lives in `NixCommandLine`, so I shouldn't need a different struct in order to perform the operation. Furthermore, you're exporting the **`nix`** config, not the `flox` config, so it makes sense to call this on the `NixCommandLine` struct instead of `flox`.

Reply:
The comment and method name at least need to be changed. From the runix perspective, there's no reason to know that `defaults.environment` contains Nix conf settings. It could contain anything

Reply:
Alternatively, I think runix could invoke every Nix related subprocess (which I would say parser-util is) with the same environment variables

Reply:
I will say though re: method name, you're unlikely to be sticking arbitrary environment variables onto the `NixCommandLine` struct, so `export_config_env_vars` at least makes the intention clear.

Reply:
> Sure, does `export_env_vars` make more sense?
Yes

> I'm not sure I follow. I haven't found a non-invasive solution to invoking subprocesses with these environment variables. Subprocesses need to either inherit the environment from their parent process, or be invoked with those environment variables set. Right now we're doing the first option because invoking commands requires passing those environment variable values to the call site, which is major surgery in both `flox` and `runix`.
> 
> If `runix` was just a module inside `flox` you could initialize a `static` at startup with the `nix` config and anything that needs it could grab it as needed. I would prefer not to do that because global variables are a code smell. In general I prefer to generate a `config` struct at startup that contains any and all settings/options you could ever need and pass that down the call stack. Doing that would require changing _a lot_ of interfaces in both `flox` and `runix` though.

I meant adding https://github.com/flox/runix/blob/d1ba2e30123f8cc340f22284df0c0c54dd2dd2b1/crates/runix/src/command_line/mod.rs#L225 where we invoke parser-util https://github.com/flox/runix/blob/d1ba2e30123f8cc340f22284df0c0c54dd2dd2b1/crates/runix/src/url_parser.rs#L492

But yes I was forgetting how painful the surgery would be as `call_bin` is not part of `NixCommandLine`

Reply:
The third alternative is a static `OnceCell` exported in runix, as in the usecase of flox we'd set that config only once.
If we do set it more often `OnceCell<Mutex<Env>>` or `OnceCell<ArcSwap<Env>>` would do the trick allowing to update the config in longer running processes (currently not really a concern i guess).
In fact using something like this by _both_ the `NixCommandLine` and `parser-util` would make the env configuration of both more uniform.

Anyway implementing that now, stretches the goals of this PR.